Use new features from deal.II [WIP]

peterrum opened the pull request:
Specifically, following (recently implemented) features are used:
- use `parallel::fullydistributed::Triangulation` 
- explicitly set length vectorization length in (`MatrixFree` and `FEEvaluation`) 

These changes allow to work on subpartitions consisting of only a single cell, which might improve the strong-scaling limit.

peterrum committed:
Use p:f:t and explicitly set length vectorization length

kronbichler commented on bp5/bench.cc:
```diff
@@ -199,7 +237,7 @@ void test(const unsigned int s,
       deallog << "True residual norm: " << tmp.l2_norm() << std::endl;
     }
-  AssertThrow(solver_control.last_step() == iterations_basic,
+  AssertThrow(true || solver_control.last_step() == iterations_basic,
```
This assertion is too unstable - would you mind removing it or relax it by requiring += 2 iterations around that value? (And for the other appearances as well)? Thanks!

peterrum replied:
I'll do!

kronbichler commented on common_code/mass_operator.h:
```diff
@@ -142,7 +142,7 @@ namespace Mass
             phi.submit_value (phi.get_value(q), q);
           phi.integrate (true,false);
-          VectorizedArray<Number> local_sum = VectorizedArray<Number>();
+          VectorizedArrayType local_sum = VectorizedArrayType();
```
Don't you need to also change the default template argument in `FEEvaluation` (here and elsewhere) to use `VectorizedArrayType`? I don't think this will compile when you enable the version `1`.

peterrum replied:
I have started with BP5, which does not need the mass operator ;) Commit comes soon!

kronbichler reviewed (changes requested):
Looks good apart from two topics listed below.

peterrum committed:
Move creation of triangulation in new function

peterrum committed:
Adjust BP1 so that it works for different vector lenghts

peterrum committed:
Use new create_triangulation in BP1

peterrum committed:
Fix compilation of other BPs

peterrum committed:
Update asserts

peterrum commented:
@kronbichler I have addressed your comments. For BP1 and BP5, we can switch between `p:f:t, n=1` and  `p:d:t, n=8`. All other BPs use the standard (old) configuration for now.
